pkg/compact/compact.go

```diff
@@ -783,7 +776,7 @@ func (cg *Group) compact(ctx context.Context, dir string, planner Planner, comp
 			return false, ulid.ULID{}, halt(errors.Wrapf(err, "block with not healthy index found %s; Compaction level %v; Labels: %v", bdir, meta.Compaction.Level, meta.Thanos.Labels))
 		}
 
-		if err := stats.OutOfOrderChunksErr(); cg.skipChunksWithOutofOrderBlocks && err != nil {
+		if err := stats.OutOfOrderChunksErr(); err != nil {
```

Comment:
With the code in `main`, if `cg.skipChunksWithOutofOrderBlocks` is `false` then we don't return error but we do `toCompactDirs = append(toCompactDirs, bdir)` below, which is wrong.